src/osd/PrimaryLogPG.cc

```diff
@@ -6275,6 +6277,8 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	  result = start_flush(ctx->op, ctx->obc, true, &missing, std::nullopt);
 	  if (result == -EINPROGRESS)
 	    result = -EAGAIN;
+    else if (result == -EINVAL)
+      goto fail;
 	} else {
```

Comment:
I don't think this is needed here because op is CEPH_OSD_OP_CACHE_FLUSH---not related to dedup work.